samples/oauth_rails/VetsApiClaims/app/views/claims/form.html.erb

```diff
@@ -34,7 +34,7 @@
 
         $('#root_authenticity_token').attr('value', $('meta[name=csrf-token]').attr('content'))
         $.post(`${window.location.pathname}/submit`, formData, function(data) {
-          window.location.href = `${window.location.pathname}/${data.id}`
+          window.location.href = `${window.location.pathname}/${data.id}?type=${data.type}&${window.location.search.substr(1)}`
```

Comment:
does there need to be an additional query param added for the `window.location.search` substring at the end? it looks like there needs to be `to=` or something similar added before `${window.location.search.substr(1)}`

Reply:
@hinzed1127 i'm not too familiar with `search.substr(1)` either, but this looks like it does what he's intending (judging from: https://stackoverflow.com/a/14395123 )

Reply:
The additional query param is just a quick way to take whatever params you have on a form and transfer them from the form to the post rather quickly.

Reply:
`search.substr(1)` makes sense to me, it just looks like it's missing the actual param name. This would generate a URL that is combining `data.type` and the search string into a single query param value, ie
```
someUrl/subpath?type=datatype&searchSubstring
```
when it seems the desired outcome is actually
```
someUrl/subpath?type=datatype&search=searchSubstring
```
I could be reading that wrong, though, if type is one large combined string and it gets parsed out elsewhere?

Reply:
It gets read by and parsed in the routes of the application itself during redirects. This allows us to quickly do a form POST for any form and then redirect to that form's show end point without needing different set of code for each form type etc.